aopy/analysis/base.py

```diff
@@ -2316,7 +2317,159 @@ def windowed_xval_lda_wrapper(data, labels, samplerate, lags=3, nfolds=5, regula
         return decoding_accuracy, time_axis, cm
     else:
         return decoding_accuracy, time_axis
+
+def xval_lda_resample_wrapper(data, labels, min_trial, cond_mask=None, single_decoding=True, min_unit=None, shuffle=False, nfolds=5,\
+                               replacement=False, regularization='auto', lda_model=None, return_weights=False, seed=None):
+    '''
+    Perform an n-fold cross-validation with Linear Discriminant Analysis (LDA) using a single unit or multiple units.
+    This functions extracts trials and/or units randomly based on the random seed with/without replacement.
+    The number of trials per target is controlled to be the same across targets
+    You can repeatedly perform this function by changing random seed to estimate the resampling distribution
+
+    Args:
+        data (nunit, ntr): neural data
+        labels (ntr): the labels for each trial.
+        min_trial (int): the minimum number of trials for each label to be extracted
+                                if this is 20, 20*(the number of unique labels) trials are extracted
+        cond_mask (ntr): a boolean array. This is a mask to extract trials that satisfies a certain condition
+                                (ex. movement onset is more than a certain threshold)
+        single_decoding (bool, optional): If True, LDA is performed using single unit activity separately (default is True)
+        min_unit (int, optional): the number of units used for decoding. This is used only when single_decoding is False.
+                                if min_unit is None, decoding is perfomed using all units (default is None)
+        shuffle (bool, optional): whether to shuffle labels or not (default is False)
+        nfolds (int, optional): the number of folds for cross-validation (default is 5)
+        replacement (bool, optional): whether to choose trials with replacement or without replacement
+        regularization (str or float, optional): If regularization should be included when building the LDA model
+                                Input into the shrinkage parameter of the sklearn LDA function
+                                Can either be None, 'auto', or a float between 0 and 1. 
+        lda_model (sklearn LDA class, optional): User-defined LDA model from sklearn.discriminant_analysis.LinearDiscriminantAnalysis. If None, this function will initialize the model.
+        return_weights (bool, optional): Whether to return the LDA weights (default is False).
+        seed (int, optional): random seed
+
+    Returns:
+        tuple: Tuple containing:
+            | **pred_Y (nch,ntr) / (ntr):** a list of predicted labels. If single_decoding is False, its shape becomes (ntr).
+            | **true_Y (nch,ntr) / (ntr):** a list of true labels. If single_decoding is False, its shape becomes (ntr).
+            | **(Optional) LDA Weights (nlabels, nch, nfolds):** The LDA weights if return_weights is True.
```

Comment:
I think this function should compute the decoding accuracy and return it. Returning the labels can be and option